From 07b69af1c431ae97c739e50a508daacc9ad38d39 Mon Sep 17 00:00:00 2001
From: Marco Farine <marco.farine@eversys.com>
Date: Tue, 21 Mar 2023 09:08:16 -0400
Subject: [PATCH] fixed 2nd seat replacing first

---
 platform/wayland/cog-platform-wl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/platform/wayland/cog-platform-wl.c b/platform/wayland/cog-platform-wl.c
index 9e13d18..d0a32e6 100644
--- a/platform/wayland/cog-platform-wl.c
+++ b/platform/wayland/cog-platform-wl.c
@@ -791,7 +791,7 @@ registry_global (void               *data,
         xdg_wm_base_add_listener(wl_data.xdg_shell, &xdg_shell_listener, NULL);
     } else if (strcmp(interface, zwp_fullscreen_shell_v1_interface.name) == 0) {
         wl_data.fshell = wl_registry_bind(registry, name, &zwp_fullscreen_shell_v1_interface, 1);
-    } else if (strcmp(interface, wl_seat_interface.name) == 0) {
+    } else if (strcmp(interface, wl_seat_interface.name) == 0 && wl_data.seat == NULL) {
         wl_data.seat = wl_registry_bind(registry, name, &wl_seat_interface, MAX(3, MIN(version, 7)));
 #if COG_ENABLE_WESTON_DIRECT_DISPLAY
     } else if (strcmp(interface, zwp_linux_dmabuf_v1_interface.name) == 0) {
